### PR TITLE
Added app uninstallation feature and make -r in installation optional

### PR DESCRIPTION
**Pull request by MohammadAG**

### CODE REVIEW

File: adb/adb_commands.py [hunk 1]
```diff
@@ -129,8 +130,30 @@ def Install(self, apk_path, destination_dir='', timeout_ms=None):
     basename = os.path.basename(apk_path)
     destination_path = destination_dir + basename
     self.Push(apk_path, destination_path, timeout_ms=timeout_ms)
-    return self.Shell('pm install -r "%s"' % destination_path,
-                      timeout_ms=timeout_ms)
+
+    cmds = ['pm install']
```
Comment (fahhem) [inline adb/adb_commands.py]: call this `cmd` since it's only one command. Or `args` or `pieces` if you want it plural

File: adb/adb_commands.py [hunk 2]
```diff
@@ -129,8 +130,30 @@ def Install(self, apk_path, destination_dir='', timeout_ms=None):
...
+
+    Args:
+      package_name: Package name of target package.
+      keep_data: whether to keep the data and cache directories
+      timeout_ms: Expected timeout for pushing and installing.
+
+    Returns:
+      The pm uninstall output.
+    """
+    cmds = ['pm uninstall']
```
Comment (fahhem) [inline adb/adb_commands.py]: same as above

**Comment by MohammadAG** [comment]
Done, no clue why I used cmds.

File: adb/adb_commands.py [hunk 3]
```diff
@@ -129,8 +130,30 @@ def Install(self, apk_path, destination_dir='', timeout_ms=None):
...
     self.Push(apk_path, destination_path, timeout_ms=timeout_ms)
-    return self.Shell('pm install -r "%s"' % destination_path,
-                      timeout_ms=timeout_ms)
+
+    cmd = ['pm install']
+    if replace_existing:
+      cmd.append('-r')
+    cmd.append('"%s"' % destination_path)
+
+    return self.Shell(" ".join(cmd), timeout_ms=timeout_ms)
```
Comment (fahhem) [inline adb/adb_commands.py]: last thing, use `'` here to be consistent with quotes

**Comment by fahhem** [comment]
Thanks!

**Pull request merged**